apis/v1alpha1/shared_types.go

```diff
@@ -67,6 +67,40 @@ type ContainerSelector struct {
 	ContainerNames *[]string `json:"containerNames,omitempty"`
 }
 
+// NetworkNamespaceSelector identifies a network namespace for network-related
+// program types in the namespace-scoped BpfApplication object.
+type NetworkNamespaceSelector struct {
+	// Target pods. This field must be specified, to select all pods use
+	// standard metav1.LabelSelector semantics and make it empty.
+	Pods metav1.LabelSelector `json:"pods"`
+}
+
+type ClPodSelector struct {
+	// Target namespaces.
+	// +optional
+	// +kubebuilder:default:=""
+	Namespace string `json:"namespace"`
+
+	// Target pods. This field must be specified, to select all pods use
+	// standard metav1.LabelSelector semantics and make it empty.
+	Pods metav1.LabelSelector `json:"pods"`
+}
+
+// ClNetworkNamespaceSelector identifies a network namespace for network-related
+// program types in the cluster-scoped ClusterBpfApplication object.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+type ClNetworkNamespaceSelector struct {
+	// Pods allows selecting the network namespace of a pod
+	// +optional
+	Pods *ClPodSelector `json:"podSelector,omitempty"`
+
+	// NamespacePath explicitly specifies a network namespace path (e.g.,
+	// /proc/<pid>/ns/net or /var/run/netns/<name>)
+	// +optional
+	NamespacePath *string `json:"namespacePath,omitempty"`
```

Comment:
why `*string` as oppose to just `string` ?

Reply:
I used the same logic for optional slices.

Reply:
I changed the *string's to just string.